File: praktikum/lib/src/sensor_data.dart
import 'dart:ffi';

import 'package:flutter/foundation.dart';

class SensorData {
  final double value;
  final DateTime timestamp;
  final SensorMetadata metadata;

  const SensorData(this.value, this.timestamp, this.metadata);

  Map<String, dynamic> toJson() => {
        'value': value,
        // Format timestamp as JSON ISO 8601 string
        'timestamp': timestamp.toUtc().toIso8601String(),
        'metadata': metadata.toJson(),
      };
}

class SensorMetadata {
  final String sensor_type;

  const SensorMetadata(this.sensor_type);

  Map<String, dynamic> toJson() => {
        'sensor_type': sensor_type,
      };
}

File: backend/mongodb/sensor_data.go
package mongodb

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Sensor_Metadata struct {
	DeviceId   string `json:"device_id,omitempty" bson:"device_id,omitempty"`
	SensorType string `bson:"type" json:"sensor_type"`
}

type Sensor_data struct {
	// id        string    `bson:"_id"`
	Value     float64            `bson:"value" json:"value"`
	Timestamp primitive.DateTime `bson:"timestamp" json:"timestamp"`
	Metadata  Sensor_Metadata    `bson:"metadata" json:"metadata"`
}

type Post_data struct {
	Value          float64            `bson:"value" json:"value"`
	Timestamp      primitive.DateTime `bson:"timestamp" json:"timestamp"`
	SensorMetadata Sensor_Metadata    `bson:"metadata" json:"metadata"`
}
